docs/SSO/Implementing-Oauth2-IDP.md

```diff
@@ -0,0 +1,94 @@
+---
+stoplight-id: z7r8pw346ahwd
+---
+
+# Implementing-Oauth2-IDP
+
+To implement the Oauth2, you need to create and configure the three endpoints.
+
+you can check the [Flow Diagram](oauth2_flow_diagram.md) which shows how Oauth2 works.
+
+**Authorization Endpoint**
+
+An authorization URL where the resource owner grants authorization to the OAuth client to access the protected resource
+
+Request Params :
+
+
+
+parameter | Description  
+---------|----------|
+|client_id    |Oauth client having the client id and client secret, you                       have to give client id.
+|response_type |Tells the authorization server which grant to execute.
+|redirect_uri  |A successful response from this endpoint results in a redirect to this URL.
+|scope |list of permissions that the application requires.
+|state |An opaque value, used for security purposes.
+
+eg.
+
+    {domain_name}/oauth2/authorize?
+    client_id=31300CDFGHRT &   
+    redirect_uri={redirect_url} &
+    response_type=code & 
+    scope=openid,email &     
+    state=f9376d0d-badd-48b4-bf8a-872978aa0098
+
+
+Response Params :
+
+parameter | Description  
+---------|----------|
+|code    | Authorization Code which is generated by server, we need to use this code for token endpoint.
+|state  |The exact value that was provided in the request
+
+
+
+**Token Endpoint**
+
+A token request URL where the OAuth client exchanges an authorization grant to generate access token.
+
+Request Params :
+
+parameter | Description  
+---------|----------|
+|client_id | The ID of the application. 
+|client_secret | password or secret code of the application.
+|grant_type   |grant_type should be authorization_code. 
+|code | Value from the Authorization endpoint’s response.
+|redirect_uri  |A successful response from this endpoint results in a redirect to this URL.
+
+eg.
+
+    {domain_name}/oauth2/token?
+      client_id=client &
+      client_secret=secret &
+      grant_type=authorization_code &   
+      redirect_uri={redirect_url} &![sso-flow-Page-2.png](../../assets/images/sso-flow-Page-2.png)
+
+      code=shddswjdj97-dbdbRRF &
+      redirect_uri={redirect_uri}
+
+
+Response Params :
+
+parameter | Description  
+---------|----------|
+|access_token | JWT token which gives you access of an application.
+|token_type | It Will always be bearer for Vendasta but could technically be other types.
```

Comment:
When referencing codestrings in line we should backquote them so that they get highlighted. 

```suggestion
|token_type | It Will always be `bearer` for Vendasta but could technically be other types.
```